src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/VirtualMachineTests.ps1

```diff
@@ -49,6 +49,11 @@ function Test-VirtualMachine
         Assert-AreEqual $p.NetworkProfile.NetworkInterfaces.Count 1;
         Assert-AreEqual $p.NetworkProfile.NetworkInterfaces[0].ReferenceUri $nicId;
 
+        $p = Add-AzureVMNetworkInterface -VM $p -Id $nicId -Primary;
+        Assert-AreEqual $p.NetworkProfile.NetworkInterfaces.Count 1;
```

Comment:
It is adding the same nic ID.  I updated the code such that if the same nic is added, it does not add and only update Primary value (if it is changed).